release_notes/fetch.py

```diff
@@ -0,0 +1,217 @@
+        return tuple(repo.iter_commits(f"{newest_tag.commit.hexsha}...{merge_commit.hexsha}"))
+
+    # otherwise the tags should be linear
+    return tuple(repo.iter_commits(f"{newest_tag.commit.hexsha}...{oldest_tag.commit.hexsha}"))
+
+
+class ReleaseNotesComponentAccess:
+    def __init__(self,
+                 component: gci.componentmodel.Component,
+                 repo_path: str,
+                 current_version: typing.Optional[semver.VersionInfo] = None):
+        self.component = component
+        self.source = cnudie.util.determine_main_source_for_component(self.component)
+
+        self.github_helper = github_helper_from_github_access(self.source.access)
+        self.git_helper = git_helper_from_github_access(self.source.access, repo_path)
+
+        # find all available versions
+        self.component_versions: dict[semver.VersionInfo, str] = {}
+        for ver in cnudie.retrieve.component_versions(self.component.name, self.component.current_repository_ctx()):
+            parsed_version = version.parse_to_semver(ver)
+            if parsed_version.prerelease:  # ignore pre-releases
+                continue
+            self.component_versions[parsed_version] = ver
+
+        if current_version is None:
+            if self.source.version is None:
+                raise ValueError(f"current_version not passed and not found in component source")
+            current_version = version.parse_to_semver(self.source.version)
+
+            # access tag from component
+            self.current_version_tag = self.git_helper.repo.tag(self.source.access.ref)
+        else:
+            # access tag from current version
+            self.current_version_tag = self.git_helper.repo.tag(self.component_versions[current_version])
+        self.current_version = current_version
+
+        if not self.current_version_tag:
+            raise RuntimeError(f"cannot find ref {self.source.access.ref} in repo")
+
+        self.previous_version = _find_previous_version(list(self.component_versions.keys()), current_version)
+        if self.previous_version is not None:
+            self.previous_version_tag = self.git_helper.repo.tag(self.component_versions[self.previous_version])
+
+        logger.debug(f"current: {self.current_version=}, {self.current_version_tag=}, " +
+                     f"previous: {self.previous_version=}, {self.previous_version_tag=}")
+
+        self.github_repo: github3.repos.Repository = self.github_helper.github.repository(
+            self.github_helper.owner,
+            self.github_helper.repository_name
+        )
+
+    def _lists_for_new_initial_release(self) -> tuple[tuple[git.Commit], tuple[git.Commit]]:
+        """ returns commits for new initial-release (all commits in repository starting from the tag)
+        """
+        # just return all commits starting from the current_version_tag
+        return tuple(self.git_helper.repo.iter_commits(self.current_version_tag)), tuple()
```

Comment:
that's awful - why?

the method's name is too generic, and misleading:

firstly, the method wil return a two-tuple containing two tuples (which contradicts the method's name)
secondly, what does that mean "lists for new initial release" / why do we need that?
thirdly, this is just a small wrapper around git_helper, w/ the special-case of passing-in "current_version_tag". why? this makes the code convoluted, and hard to understand